src/main/core/support/configuration.rs

```diff
@@ -707,13 +707,55 @@ impl std::ops::Deref for Quantity {
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+#[derive(Debug, Clone, Serialize, JsonSchema)]
 #[serde(untagged)]
 pub enum ProcessArgs {
     List(Vec<String>),
     Str(String),
 }
 
+/// Serde doesn't provide good deserialization error messages for untagged enums, so we implement
+/// our own. For example, if serde finds a yaml value such as 4 for the process arguments, it won't
+/// deserialize it to the string "4" and the yaml parsing will fail. The serde-generated error
+/// message will say something like "data did not match any variant of untagged enum ProcessArgs at
+/// line X column Y" which isn't very helpful to the user, so here we try to give a better error
+/// message.
+impl<'de> serde::Deserialize<'de> for ProcessArgs {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        struct ProcessArgsVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for ProcessArgsVisitor {
+            type Value = ProcessArgs;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a string or a sequence of strings")
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
```

Comment:
Possibly orthogonal to this PR, but would it be simpler to force `ProcessArgs` to always be a list? e.g. a single string value would become a list of size 1 ...?

Reply:
The main reason we support a list is to make it easier to distinguish between argv values as passed during exec. If we put the entire string in a single list item, then we would call the plugin with an argv of length 2 (process name and single argument). Right now, we use `parse_string_as_args()` later to split the string into a list following the usual shell parsing rules. But we could add the `parse_string_as_args()` call directly into this `Deserialize` impl.

Reply:
Ah I see. Yeah, moving the string-splitting here might make sense, but up to you.

I'd also be a little tempted to only support a list in the yaml to avoid having our own implementation of shell parsing rules, but it'd definitely be a bit more verbose in the config files, and maybe not worth changing...

Reply:
Sounds good. No strong opinion on whether it's better to split here or later - totally up to you